src/main/scala/com/localytics/sbt/s3/S3ProxyUtils.scala

```diff
@@ -0,0 +1,46 @@
+package com.localytics.sbt.s3
+
+import scala.util.Try
+
+private[s3] object S3ProxyUtils {
+
+  private val ProcessIDRegex = """\d+ s3proxy""".r
+
+  def extractS3ProxyPid(input: String): Option[String] = ProcessIDRegex.findFirstIn(input).map(_.split(" ")(0))
+
+  def isS3ProxyRunning(port: Int): Boolean = {
+    Try {
+      val socket = new java.net.Socket("localhost", port)
+      socket.close()
+    }.isSuccess
+  }
+
+  def waitForS3Proxy(port: Int, infoPrintFunc: String => Unit): Unit = {
+    var continue = false
+    while (!continue) {
+      continue = true
+      Try {
```

Comment:
It's a little cleaner to use `isS3ProxyRunning` in your while. Avoids the duplicate code and var.